modules/apps/transfer/types/transfer_authorization.go

```diff
@@ -37,59 +41,72 @@ func (TransferAuthorization) MsgTypeURL() string {
 }
 
 // Accept implements Authorization.Accept.
-func (a TransferAuthorization) Accept(ctx context.Context, msg proto.Message) (authz.AcceptResponse, error) {
+func (a TransferAuthorization) Accept(goCtx context.Context, msg proto.Message) (authz.AcceptResponse, error) {
 	msgTransfer, ok := msg.(*MsgTransfer)
 	if !ok {
 		return authz.AcceptResponse{}, errorsmod.Wrap(ibcerrors.ErrInvalidType, "type mismatch")
 	}
 
-	for index, allocation := range a.Allocations {
-		if !(allocation.SourceChannel == msgTransfer.SourceChannel && allocation.SourcePort == msgTransfer.SourcePort) {
-			continue
-		}
+	index := getAllocationIndex(*msgTransfer, a.Allocations)
+	if index == allocationNotFound {
+		return authz.AcceptResponse{}, errorsmod.Wrapf(ibcerrors.ErrNotFound, "requested port and channel allocation does not exist")
+	}
 
-		if !isAllowedAddress(sdk.UnwrapSDKContext(ctx), msgTransfer.Receiver, allocation.AllowList) {
-			return authz.AcceptResponse{}, errorsmod.Wrap(ibcerrors.ErrInvalidAddress, "not allowed receiver address for transfer")
-		}
+	allocation := a.Allocations[index]
+	ctx := sdk.UnwrapSDKContext(goCtx)
 
-		err := validateMemo(sdk.UnwrapSDKContext(ctx), msgTransfer.Memo, allocation.AllowedPacketData)
-		if err != nil {
-			return authz.AcceptResponse{}, err
-		}
+	if !isAllowedAddress(ctx, msgTransfer.Receiver, allocation.AllowList) {
+		return authz.AcceptResponse{}, errorsmod.Wrap(ibcerrors.ErrInvalidAddress, "not allowed receiver address for transfer")
+	}
+
+	err := validateMemo(ctx, msgTransfer.Memo, allocation.AllowedPacketData)
+	if err != nil {
+		return authz.AcceptResponse{}, err
+	}
 
+	// bool flag to see if we have updated any of the allocations
+	allocationModified := false
+
+	// update spend limit for each token in the MsgTransfer
+	for _, coin := range msgTransfer.GetCoins() {
 		// If the spend limit is set to the MaxUint256 sentinel value, do not subtract the amount from the spend limit.
-		if allocation.SpendLimit.AmountOf(msgTransfer.Token.Denom).Equal(UnboundedSpendLimit()) {
-			return authz.AcceptResponse{Accept: true, Delete: false, Updated: nil}, nil
+		// if there is no unlimited spend, then we need to subtract the amount from the spend limit to get the limit left
+		if allocation.SpendLimit.AmountOf(coin.Denom).Equal(UnboundedSpendLimit()) {
+			continue
 		}
 
-		limitLeft, isNegative := allocation.SpendLimit.SafeSub(msgTransfer.Token)
+		limitLeft, isNegative := a.Allocations[index].SpendLimit.SafeSub(coin)
```

Comment:
```suggestion
		limitLeft, isNegative := allocation.SpendLimit.SafeSub(coin)
```
take it or leave it, just felt odd relative to line 74